### PR TITLE
optimize RNN

### PR DESCRIPTION
**Pull request by QiJune**
The optimization result for the first batch training(**please remove shuffle of training data to get the accurate testing result**) is following:

| policy          | memory usage     | 
|--------------- | --------- |
|without any optimization| 5728446208 bytes|
|delete step scope| 1126832384 bytes|
|with memory reuse policy |1115982080 bytes|

Command:
```
FLAGS_benchmark=true GLOG_vmodule=executor=2 GLOG_logtostderr=0 GLOG_log_dir="./" python machine_translation.py
```

### CODE REVIEW

File: fluid/machine_translation.py [hunk 1]
```diff
@@ -256,9 +256,15 @@ def train():
     optimizer = fluid.optimizer.Adam(learning_rate=args.learning_rate)
     optimizer.minimize(avg_cost)
 
+    fluid.memory_optimize(fluid.default_main_program())
+
+    # train_batch_generator = paddle.batch(
```
Comment (dzhwinter) [inline fluid/machine_translation.py]: remove this comment out lines

File: fluid/machine_translation.py [hunk 2]
```diff
@@ -256,9 +256,15 @@ def train():
...
+    #     paddle.reader.shuffle(
+    #         paddle.dataset.wmt14.train(args.dict_size), buf_size=1000),
+    #     batch_size=args.batch_size)
+
     train_batch_generator = paddle.batch(
-        paddle.reader.shuffle(
-            paddle.dataset.wmt14.train(args.dict_size), buf_size=1000),
+            paddle.dataset.wmt14.train(args.dict_size),
```
Comment (dzhwinter) [inline fluid/machine_translation.py]: Why did we need to remove the shuffle? Or it just for testing?

Comment (QiJune) [reply fluid/machine_translation.py]: Yes, just for testing. We should fix the data order to get accurate result

File: fluid/machine_translation.py (hunk 1 above)
Comment (QiJune) [reply fluid/machine_translation.py]: I will remove it.
